src/Exceptions/HttpException.php

```diff
@@ -18,28 +18,42 @@ abstract class HttpException extends Exception
 
     public function __construct(public readonly ServerRequestInterface $request)
     {
-        parent::__construct(static::DESCRIPTION . " " . static::HREF, static::CODE);
+        parent::__construct(static::getDescription() . " " . static::HREF, static::CODE);
     }
 
     public static function buildEmptyResponse(): EmptyResponse
     {
         return new EmptyResponse(static::CODE);
     }
 
-    public static function buildHtmlResponse(): HtmlResponse
+    public static function getDescription(): string
+    {
+        return str_replace("\n", "", static::DESCRIPTION);
+    }
+
+    protected static function checkConstants(): void
     {
         $code = static::CODE;
-        $description = static::DESCRIPTION;
-        $phrase = static::PHRASE;
+        $description = trim(static::getDescription());
+        $phrase = trim(static::PHRASE);
         $href = static::HREF;
 
         if (empty($phrase)) {
             throw new RuntimeException("Phrase constant is not defined.");
         }
 
-        if ($description === '') {
+        if (empty($description)) {
             throw new RuntimeException("Description constant defined.");
         }
+    }
+
+    public static function buildHtmlResponse(): HtmlResponse
+    {
+        self::checkConstants();
+        $code = static::CODE;
+        $description = trim(static::getDescription());
```

Comment:
You trim the description everywhere, why not just, return it trimmed?

Reply:
Check newer commits, this should be fixed